mm2src/coins/lp_coins.rs

```diff
@@ -2152,39 +2158,110 @@ impl CoinsContext {
         })))
     }
 
-    pub async fn add_coin(&self, coin: MmCoinEnum) -> Result<(), MmError<CoinIsAlreadyActivatedErr>> {
+    pub async fn add_token(&self, coin: MmCoinEnum) -> Result<(), MmError<RegisterCoinError>> {
         let mut coins = self.coins.lock().await;
         if coins.contains_key(coin.ticker()) {
-            return MmError::err(CoinIsAlreadyActivatedErr {
-                ticker: coin.ticker().into(),
+            return MmError::err(RegisterCoinError::CoinIsInitializedAlready {
+                coin: coin.ticker().into(),
             });
         }
+        let ticker = coin.ticker();
+
+        let mut platform_coin_tokens = self.platform_coin_tokens.lock();
+        // Here, we tried to add to a token to platform_coin_tokens if the token belongs to a platform coin.
+        if let Some(platform) = platform_coin_tokens.get_mut(coin.platform_ticker()) {
+            platform.insert(ticker.to_owned());
+        }
 
-        coins.insert(coin.ticker().into(), coin);
+        coins.insert(ticker.into(), coin);
         Ok(())
     }
 
+    /// Adds a Layer 2 coin that bases on another standalone platform.
```

Comment:
Typo, should be `/// Adds a Layer 2 coin that is based on another standalone platform.` or change it to `/// Adds a Layer 2 coin that depends on a standalone platform.`